Florida_Analysis_Py_File.py

```diff
@@ -0,0 +1,534 @@
+deaths.head(3)
+
+deaths.columns
+# creating a copy of deaths dataset, to prevent SettingWithCopy warnings. The irrelevant columns are dropped.
+
+deaths_copy = deaths.copy()
+deaths_copy.drop(
+    [
+        "County Code",
+        "Year Code",
+        "Drug/Alcohol Induced Cause Code",
+        "State",
+        "CountyName",
+        "StateAbbr",
+        "STATE_COUNTY",
+        "county_test",
+    ],
+    axis=1,
+    inplace=True,
+)
+deaths_copy.head(3)
+# converting some of the columns to more appropriate data type - integer data type
+deaths_copy["Year"] = deaths_copy["Year"].astype("int64")
+deaths_copy["Deaths"] = deaths_copy["Deaths"].astype("int64")
+deaths_copy["Population"] = deaths_copy["Population"].astype("int64")
+
+
+# renaming some of the columns to appear more conventional
+deaths_copy.rename(
+    columns={
+        "StateFIPS": "State FIPS",
+        "CountyFIPS": "County FIPS",
+        "StateName": "State",
+        "state_abbrev": "State Abbreviation",
+        "FIP_unique": "FIPS_Unique",
+    },
+    inplace=True,
+)
+
+deaths_copy.head(3)
+deaths_copy["State"].unique()
+
+deaths_copy["Drug/Alcohol Induced Cause"].unique()
+# creating the dataset for Florida which has all deaths that are drug related
+
+florida_deaths = deaths_copy[
+    (deaths_copy["State"] == "Florida")
+    & (
+        (
+            deaths_copy["Drug/Alcohol Induced Cause"]
+            == "Drug poisonings (overdose) Unintentional (X40-X44)"
+        )
+        | (
+            deaths_copy["Drug/Alcohol Induced Cause"]
+            == "Drug poisonings (overdose) Undetermined (Y10-Y14)"
+        )
+        | (
+            deaths_copy["Drug/Alcohol Induced Cause"]
+            == "Drug poisonings (overdose) Suicide (X60-X64)"
+        )
+    )
+]
+
+florida_deaths.head(5)
+florida_deaths_copy = florida_deaths.copy()
+# calculating the drug related death rates in Florida per 100000 people
+
+florida_deaths_copy["Overdose_Per_100K"] = (
+    florida_deaths_copy["Deaths"] / florida_deaths_copy["Population"]
+) * 100_000
+
+florida_deaths_copy.head(3)
+# calculating and displaying year and county wise results for drug related deaths in Florida
+
+florida_result = (
+    florida_deaths_copy.groupby(["Year", "County"])["Overdose_Per_100K"]
+    .sum()
+    .reset_index()
+)
+florida_result.head(5)
+# calculating and displaying yearwise results for drug related deaths in Florida
+
+florida_results_yearwise = florida_deaths_copy.groupby("Year")[
+    "Overdose_Per_100K"
+].mean()
+florida_results_yearwise
+
+florida_result_copy = florida_result.copy()
+
+# create a scale for number of years before and after 2010 (target year)
+def scale_years(year):
+    if year == 2007:
+        return -3
+    if year == 2008:
+        return -2
+    if year == 2009:
+        return -1
+    if year == 2010:
+        return 0
+    if year == 2011:
+        return 1
+    if year == 2012:
+        return 2
+    if year == 2013:
+        return 3
```

Comment:
Hey Annapurna, I think we should delete this since we should present the actual years not the years relative to the policy year.

Reply:
Oops! I will make the change! Thanks Zhonglin!